py/compile.c

```diff
@@ -3600,8 +3602,12 @@ mp_compiled_module_t mp_compile_to_raw_code(mp_parse_tree_t *parse_tree, qstr so
             }
 
             // final pass: emit code
+            // the emitter can request multiple of these passes
             if (comp->compile_error == MP_OBJ_NULL) {
-                compile_scope(comp, s, MP_PASS_EMIT);
+                bool pass_complete;
```

Comment:
For some reason, I don't trust the compiler to optimize out the local variable, so, to potentially save a few bytes, would write the loop:

```c
while (!compile_scope(...)) {
}
```

Reply:
That is much better!  Done.